### PR TITLE
add yamllint

### PR DESCRIPTION
**Pull request by gdams**
The Ansible playbooks are long overdue having some linting run on them, this PR adds yamllint (will require travis to be enabled on this repo).

Key changes:

- Enforce the use of `true`/`false` rather than `yes`/`no`
- Enforce spaces between braces (`{ blah }` rather than `{blah}`)
- Enforce `---` at the top of all yaml files

### CODE REVIEW

**Comment by gdams** [comment]
@nodejs/build could I get a few more reviews before landing this please?

**Comment by gdams** [comment]
@phillipj good spot! updated PTAL

File: ansible/roles/visual-studio/tasks/partials/vs2015.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@
   win_get_url:
     url: 'https://go.microsoft.com/fwlink/?LinkId=532606'
     dest: 'C:\TEMP\vs2015_community.exe'
-    force: no
+    force: false
```
Comment (joaocgreis) [inline ansible/roles/visual-studio/tasks/partials/vs2015.yml]: Indentation updates and `---` at the top sound great to me, but text should not change. Take this line for example, the Ansible documentation at https://docs.ansible.com/ansible/latest/modules/win_get_url_module.html clearly shows that the choices are `yes` and `no`.

Comment (gdams) [reply ansible/roles/visual-studio/tasks/partials/vs2015.yml]: so `yes` and `no` are always interchangeable with `True` and `False` and ansible is pushing for defaulting to `True`/`False` in all newer versions. I'm open to either but I feel that we should agree on a format to use

File: yamllint.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+extends: default
+rules:
+  line-length: disable
+  braces: { min-spaces-inside: 1, max-spaces-inside: 1 }
+  brackets: { min-spaces-inside: 1, max-spaces-inside: 1 }
```
Comment (smpeters) [inline yamllint.yml]: If there are still some questions on true/false vs. yes/no, then maybe ignoring these issues would be the best for now.

```suggestion
  brackets: { min-spaces-inside: 1, max-spaces-inside: 1 }
  truthy: disable
```

**Comment by sam-github** [comment]
Maybe worth resurrecting this?

**Comment by richardlau** [comment]
> 
> 
> Maybe worth resurrecting this?

I'll resurrect this using GitHub Actions (assigned myself for now but I'll probably close and open a new PR).

**Comment by sam-github** [comment]
Tried it out, mostly cosmetic or confusing comments from yamllint. Closing (though if anyone wants to take another shot, go for it).

**Pull request closed without merging**